iceoryx_examples/icedelivery/iox_publisher.cpp

```diff
@@ -86,7 +86,7 @@ int main()
         {
             auto error = result.get_error();
             // Ignore unused variable warning
-            (void)error;
+            IOX_DISCARD_RESULT(error);
```

Comment:
How about just printing the enum value as integer?

The same could be done 17 lines above